sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala

```diff
@@ -112,12 +112,14 @@ case class Collate(child: Expression, collationName: String)
   since = "4.0.0",
   group = "string_funcs")
 // scalastyle:on line.contains.tab
-case class Collation(child: Expression) extends UnaryExpression with RuntimeReplaceable {
-  override def dataType: DataType = StringType
+case class Collation(child: Expression)
+  extends UnaryExpression with RuntimeReplaceable with ExpectsInputTypes {
+  override def dataType: DataType = child.dataType
```

Comment:
shouldn't the collation name be a normal string type?

Reply:
I was thinking of some cases when indeterminate comes. What if our costumer wants to concat the collation used to their strings and filter on something like that. If we return a StringType, we would face a problem of getting indeterminate collation failing to keep the pass through for our functions. This is more of a problem if users set default collation to be something else, so we might say that changing return type to default collation would be the better choice?